server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1589,6 +1509,148 @@ static String resolveExpression(String expression, LongSupplier getTime) {
         }
     }
 
+    public static final class ExplicitResourceNameFilter {
+
+        private ExplicitResourceNameFilter() {
+            // Utility class
+        }
+
+        public static List<String> filter(Context context, List<String> expressions) {
+            List<String> result = new ArrayList<>(expressions.size());
+            for (ExpressionIterable.Expression expression : new ExpressionIterable(context, expressions)) {
+                validateAliasOrIndex(expression);
```

Comment:
This is arguably not the best place to do this validation but it's closest to where the original code does it...
Open to suggestions on where to move this, if anyone has opinions on it.